homeassistant/components/tplink/config_flow.py

```diff
@@ -137,16 +345,109 @@ def _async_create_entry_from_device(self, device: SmartDevice) -> FlowResult:
             title=f"{device.alias} {device.model}",
             data={
                 CONF_HOST: device.host,
+                CONF_ALIAS: device.alias,
+                CONF_MODEL: device.model,
+                CONF_CONNECTION_TYPE: device.config.connection_type.to_dict(),
```

Comment:
I think the plan was to use `DeviceConfig` as is, e.g., `CONF_CONFIG` instead of doing it like this?

Reply:
I'm still not convinced we should be storing the entire `DeviceConfig` in HA per config entry.  Other than the connection type and the host all the other values are "global" values, i.e. timeouts and credentials etc that we would expect consumers to set themselves.  It's only really the host and the connection type that's needed for `Connect()`.  

Also I was thinking of adding an option to the credentials to allow a user to select whether to persist the credentials or not.  That said it was a bit annoying that I had to start persisting `uses_http` separately as well so I was thinking of either moving `uses_http` to the connection type or having `to_dict()` on `DeviceConfig` take a parameter of `include_credentials`.  What do you think?

Reply:
Hey @rytilahti could you please opine on this question before I make any changes. Many thanks.

Reply:
Do you have specific concerns on using the entire device config? Or what could convince you otherwise? IMO, the downstream users of the library API should not need to be concerned on internal workings of the library to be able to use it effectively. 

Effectively for the use case at hand:
1. Perform discovery or setup the device otherwise with the wanted params.
2. Make it possible to serialize the working configuration for later use.
3. Re-initialize the same device instance from the serialized object.

Some thoughts:
* Could you remind me what's the problem with uses_http etc. you mention in the second paragraph?
* If storing the credentials in clear text is an issue, we could instead export the hashed versions and use those directly?